aten/src/ATen/DLConvertor.cpp

```diff
@@ -129,7 +130,7 @@ static Device getATenDevice(const DLDevice& ctx, void* data) {
       return at::Device(DeviceType::HIP, ctx.device_id);
 #endif
     case DLDeviceType::kDLOneAPI:
-      return at::detail::getXPUHooks().getATenDeviceFromDLPackDevice(ctx, data);
+      return at::detail::getXPUHooks().getDeviceFromPtr(data);
```

Comment:
Can you explain why we cannot do `at::Device(DeviceType::XPU, ctx.device_id)` and has to be different from other devices here?

Reply:
The main reason is that the SYCL specification does not explicitly state that the order of enumeration for SYCL devices is fixed. The external program may not necessarily correspond to the same order as device index in PyTorch. So we have to be cautious about this and find the target device via data pointer.
Please correct me if I am wrong.@gujinghui